sdcm/cluster_aws.py

```diff
@@ -513,6 +513,18 @@ def external_address(self):
         else:
             return self._instance.private_ip_address
 
+    @cached_property
```

Comment:
I wonder if we need to uncache it on reboot of a node

Reply:
in that case we should raise some issue - as it would break cluster network.

Reply:
seems like we are using by default the IP name 
```
Hostname type
IP name: ip-10-4-0-64.eu-west-1.compute.internal
```
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-naming.html

there is one case where we stop and start instance, it's in `AWSNode.restart()` it's part of nemesis that is long disabled.
but we are calling there `refresh_ip_address`, that clears all ip related caches, since we've seen cases the public ip address is changing, I hope that local ip address never changes, cause our hostname are matching it.

I still think we should clear the cache in `refresh_ip_address` calls, and maybe generate error event if it was changing.

Reply:
regardless, dns name should never change

Reply:
in that case, it may happen. Added proper cache invalidation in `refresh_ip_address` (btw. fixing `cql_ip_address` invalidation along)